### PR TITLE
LPS-104365 Listen to ResourceBundles changes in order to change modificationDate since JS can change if it's using ResourceBundle keys.

### PR DESCRIPTION
**Pull request by marianoalvarosaiz**

### CODE REVIEW

**Comment by shuyangzhou** [comment]
ci:test

File: modules/apps/frontend-js/frontend-js-web/src/main/java/com/liferay/frontend/js/web/internal/JavaScriptPortalWebResources.java [hunk 1]
```diff
@@ -65,6 +83,18 @@ protected void setServletContext(ServletContext servletContext) {
...
+	private AtomicLong _lastModified;
+	private final ResourceBundleServiceListener _resourceBundleServiceListener =
+		new ResourceBundleServiceListener();
 	private ServletContext _servletContext;
 
+	private class ResourceBundleServiceListener implements ServiceListener {
+
+		@Override
+		public void serviceChanged(ServiceEvent serviceEvent) {
+			_lastModified.incrementAndGet();
```
Comment (shuyangzhou) [inline modules/apps/frontend-js/frontend-js-web/src/main/java/com/liferay/frontend/js/web/internal/JavaScriptPortalWebResources.java]: @marianoalvarosaiz I doubt this is right. The getLastModified() use this timestamp to build URLs return to client. It was purely based on the bundle updates, now with this change, the timestamp can "randomly" bump between starts. Meaning the client "bookmarked" url no longer knows it is up to date or not.

How is the resource bundle related to this last modified time? It is not describe in the ticket.

And all the servicelisteners are notified tons of times during startup, due to we have a lot buildin ResourceBundle published. This is going to impact startup time. It must be avoid. You just want to react for runtime deployed resource bundles, not those buildin ones during startup.

Please solve these issues and update the tickets with reproducible steps for each case you are fixing. And upload all test wars you use.

**Comment by shuyangzhou** [comment]
@marianoalvarosaiz please see my comment above, fix those issues and update ticket, then resend.

**Comment by koor4356** [comment]
@marianoalvarosaiz,  I am seeing this "StagingUsecase#ImportLARsOnStagedSite" fails on your pull request. I was able to reproduce when I run an automation test locally. Can please you look into this?
`01:36:08,235 ERROR [http-nio-8080-exec-3][AssetPublisherPortletToolbarContributor:178] Unable to add folder menu item
     [exec] java.lang.NullPointerException
     [exec] 	at com.liferay.asset.internal.util.AssetHelperImpl.getAssetPublisherAddItemHolders(AssetHelperImpl.java:318)
     [exec] 	at com.liferay.asset.publisher.web.internal.display.context.AssetPublisherDisplayContext.getScopeAssetPublisherAddItemHolders(AssetPublisherDisplayContext.java:1069)
     [exec] 	at com.liferay.asset.publisher.web.internal.portlet.toolbar.contributor.AssetPublisherPortletToolbarContributor.addPortletTitleAddAssetEntryMenuItems(AssetPublisherPortletToolbarContributor.java:92)
     [exec] 	at com.liferay.asset.publisher.web.internal.portlet.toolbar.contributor.AssetPublisherPortletToolbarContributor.getPortletTitleMenuItems(AssetPublisherPortletToolbarContributor.java:174)
     [exec] 	at com.liferay.portal.kernel.portlet.toolbar.contributor.BasePortletToolbarContributor.getPortletTitleMenus(BasePortletToolbarContributor.java:55)
     [exec] 	at com.liferay.portal.kernel.portlet.toolbar.PortletToolbar.getPortletTitleMenus(PortletToolbar.java:101)`

I am having a hard time reproducing it manually, but I will keep digging and see if I can get the steps manually.

Thank you!

**Comment by marianoalvarosaiz** [comment]
see https://github.com/shuyangzhou/liferay-portal/pull/8665

**Pull request closed without merging**